src/app/@ansyn/ansyn/app-effects/effects/tools.app.effects.ts

```diff
@@ -186,62 +126,34 @@ export class ToolsAppEffects {
 		.map((action) => new SetActiveOverlaysFootprintModeAction(action.payload));
 
-	/**
-	 * @type Effect
-	 * @name onDisplayOverlaySuccess$
-	 * @ofType DisplayOverlaySuccessAction
-	 * @dependencies map
-	 * @action EnableImageProcessing, SetMapAutoImageProcessing?, SetMapManualImageProcessing?, SetManualImageProcessingArguments, SetAutoImageProcessingSuccess
-	 */
 
 	@Effect()
 	onDisplayOverlaySuccess$: Observable<any> = this.actions$
 		.ofType<DisplayOverlaySuccessAction>(OverlaysActionTypes.DISPLAY_OVERLAY_SUCCESS)
-		.withLatestFrom(this.store$.select(casesStateSelector), this.store$.select(mapStateSelector),
-			(action: DisplayOverlaySuccessAction, cases: ICasesState, mapsState: IMapState) => {
-				const mapId = action.payload.mapId || mapsState.activeMapId;
-				const selectedMap: CaseMapState = MapFacadeService.mapById(mapsState.mapsList, mapId);
-				return [action, selectedMap, cloneDeep(cases.selectedCase)];
-			})
-		.filter(([action, selectedMap, selectedCase]: [DisplayOverlaySuccessAction, CaseMapState, Case]) => Boolean(selectedMap))
-		.mergeMap(([action, selectedMap, selectedCase]: [DisplayOverlaySuccessAction, CaseMapState, Case]) => {
-			// action 1: EnableImageProcessing
-			const actions = [new EnableImageProcessing()];
-			let manualProcessArgs: ImageManualProcessArgs;
-			let params = this.config.ImageProcParams.map(param => {
-				return param.defaultValue;
-			});
+		.withLatestFrom(this.store$.select(mapStateSelector), this.store$.select(toolsStateSelector))
+		.mergeMap(([action, mapState, toolsState]: [DisplayOverlaySuccessAction, IMapState, IToolsState]) => {
+			const imageManualProcessArgs: ImageManualProcessArgs = this.defaultImageManualProcessArgs;
 
-			manualProcessArgs = {
-				Sharpness: params[0],
-				Contrast: params[1],
-				Brightness: params[2],
-				Gamma: params[3],
-				Saturation: params[4]
-			};
+			const actions = [];
+			const updatedMapList = [...mapState.mapsList];
 
-			// action 2: SetMapManualImageProcessing / SetMapAutoImageProcessing (optional)
-			if (selectedCase.state.overlaysManualProcessArgs) {
-				manualProcessArgs = selectedCase.state.overlaysManualProcessArgs[action.payload.overlay.id] || manualProcessArgs;
-			}
-			if (selectedMap.data.isAutoImageProcessingActive) {
-				// auto process action
-				actions.push(new SetMapAutoImageProcessing({
-					mapId: selectedMap.id,
-					toggleValue: selectedMap.data.isAutoImageProcessingActive
-				}));
-			} else if (manualProcessArgs) {
-				// manual process action
-				actions.push(new SetMapManualImageProcessing({
-					mapId: selectedMap.id,
-					processingParams: manualProcessArgs
-				}));
+			const mapToUpdate = updatedMapList.find((map) => map.id === MapFacadeService.activeMap(mapState).id);
+			mapToUpdate.data.imageManualProcessArgs = toolsState.overlaysManualProcessArgs[action.payload.overlay.id];
+			actions.push(new SetMapsDataActionStore({ mapsList: updatedMapList }));
+			return actions;
```

Comment:
return single action `map` instead of  `mergeMap`

Reply:
done